iothub/device/src/Logger/EventCounterLogger.cs

```diff
@@ -0,0 +1,347 @@
+        /// </summary>
+        void OnAmqpConnectionDisconnected();
+        /// <summary>
+        /// When AMQP session was established.
+        /// </summary>
+        void OnAmqpSessionEstablished();
+        /// <summary>
+        /// When AMQP session was disconnected.
+        /// </summary>
+        void OnAmqpSessionDisconnected();
+        /// <summary>
+        /// When AMQP token refresher was started.
+        /// </summary>
+        void OnAmqpTokenRefresherStarted();
+        /// <summary>
+        /// When AMQP token refresher was stopped.
+        /// </summary>
+        void OnAmqpTokenRefresherStopped();
+        /// <summary>
+        /// When AMQP token was refreshed.
+        /// </summary>
+        void OnAmqpTokenRefreshed();
+        /// <summary>
+        /// Enable logger and start logging
+        /// </summary>
+        /// <param name="interval">The interval of each logging period.</param>
+        /// <param name="redirectToConsole">Dump log to file.</param>
+        /// <param name="cancellationToken">Cancel the loop.</param>
+        Task StartLoggerAsync(TimeSpan interval, bool redirectToConsole, CancellationToken cancellationToken);
+    }
+
+#if NETSTANDARD2_0
+    /// <summary>
+    /// Logger for device client events.
+    /// </summary>
+    [EventSource(Name = "Microsoft-Azure-Devices-Client-Logger-Event-Counter")]
+    public class EventCounterLogger : EventSource, IEventCounterLogger
+    {
+        private static EventCounterLogger s_instance = new EventCounterLogger();
+
+        
+
+        private bool _started;
+        private int _deviceClientCreationCounts;
+        private int _deviceClientDisposeCounts;
+        private int _amqpUnitCreationCounts;
+        private int _amqpUnitDisposeCounts;
+        private int _amqpConnectionEstablishCounts;
+        private int _amqpConnectionDisconnectionCounts;
+        private int _amqpSessionEstablishCounts;
+        private int _amqpSessionDisconnectionCounts;
+        private int _amqpTokenRefreshStartCounts;
+        private int _amqpTokenRefreshStopCounts;
+        private int _amqpTokenRefreshCounts;
+
+        /// <summary>
+        /// Return the instance.
+        /// </summary>
+        public static IEventCounterLogger GetInstance()
+        {
+            return s_instance;
+        }
+
+        private EventCounterLogger() : base()
+        {
+        }
+
+        /// <summary>
+        /// When device client was created.
+        /// </summary>
+        public void OnDeviceClientCreated()
+        {
+            if (_started)
+            {
+                Interlocked.Increment(ref _deviceClientCreationCounts);
+            }
+        }
+
+        /// <summary>
+        /// When device client was disposed.
+        /// </summary>
+        public void OnDeviceClientDisposed()
+        {
+            if (_started)
+            {
+                Interlocked.Increment(ref _deviceClientDisposeCounts);
```

Comment:
I don't understand this metric: all of them seem to always increment. Shouldn't the DeviceCreated be paired with Device Disposed to show fluctuations in how many clients are active at a time?